cli/pkg/kctrl/cmd/app/kick.go

```diff
@@ -0,0 +1,172 @@
+// Copyright 2020 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package app
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/cppforlife/go-cli-ui/ui"
+	"github.com/spf13/cobra"
+	cmdcore "github.com/vmware-tanzu/carvel-kapp-controller/cli/pkg/kctrl/cmd/core"
+	"github.com/vmware-tanzu/carvel-kapp-controller/cli/pkg/kctrl/logger"
+	kcv1alpha1 "github.com/vmware-tanzu/carvel-kapp-controller/pkg/apis/kappctrl/v1alpha1"
+	kcclient "github.com/vmware-tanzu/carvel-kapp-controller/pkg/client/clientset/versioned"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+type KickOptions struct {
+	ui          ui.UI
+	depsFactory cmdcore.DepsFactory
+	logger      logger.Logger
+
+	WaitFlags cmdcore.WaitFlags
+
+	NamespaceFlags cmdcore.NamespaceFlags
+	Name           string
+}
+
+func NewKickOptions(ui ui.UI, depsFactory cmdcore.DepsFactory, logger logger.Logger) *KickOptions {
+	return &KickOptions{ui: ui, depsFactory: depsFactory, logger: logger}
+}
+
+func NewKickCmd(o *KickOptions, flagsFactory cmdcore.FlagsFactory) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "kick",
+		Short: "Trigger reconciliation for App CR",
+		RunE:  func(_ *cobra.Command, _ []string) error { return o.Run() },
+	}
+
+	o.NamespaceFlags.Set(cmd, flagsFactory)
+	cmd.Flags().StringVarP(&o.Name, "app", "a", "", "Set App CR name (required)")
+	o.WaitFlags.Set(cmd, flagsFactory, &cmdcore.WaitFlagsOpts{
+		AllowDisableWait: true,
+		DefaultInterval:  1 * time.Second,
+		DefaultTimeout:   5 * time.Minute,
+	})
+
+	return cmd
+}
+
+func (o *KickOptions) Run() error {
+	if len(o.Name) == 0 {
+		return fmt.Errorf("Expected App CR name to be non empty")
```

Comment:
do we use `CR` in desriptions/messages? (e.g. is it PackageInstall CR in other places)